File: 백준/Gold/5052. 전화번호 목록/전화번호 목록.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
)

var reader = bufio.NewReader(os.Stdin)

func calculate(n int) {
	var ret = "YES"
	var strs = make([]string, n)
	var breakFlag = false
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%s\n", &strs[i])
	}
	sort.Strings(strs)

	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			if strings.HasPrefix(strs[j], strs[i]) {
				ret = "NO"
				breakFlag = true
				break
			}
		}
		if breakFlag {
			break
		}
	}
	fmt.Println(ret)

}
func main() {
	var tc, n int
	fmt.Fscanf(reader, "%d\n", &tc)
	for i := 0; i < tc; i++ {
		fmt.Fscanf(reader, "%d\n", &n)
		calculate(n)
	}
}

File: 백준/Gold/5052. 전화번호 목록/전화번호 목록.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)

def cal(n):
    strs = []
    for _ in range(n):
        strs.append(input().strip('\n'))
    strs.sort()
    ret = "YES"
    break_flag = False
    for i in range(0, n-1):
        if len(strs[i+1])>=len(strs[i]) and strs[i+1][:len(strs[i])]==strs[i]:
            ret = "NO"
            break
    print(ret)    
tc= int(input())
for i in range(tc):
    cal(int(input()))